[CI:DOCS] podman-generate-systemd.1.md: document --sdnotify

eriksjolund opened the pull request:
Fixes #15029

Signed-off-by: Erik Sjölund <erik.sjolund@gmail.com>

<!--
Thanks for sending a pull request!

Please make sure you've read our contributing guidelines and how to submit a pull request (https://github.com/containers/podman/blob/main/CONTRIBUTING.md#submitting-pull-requests).

In case you're only changing docs, make sure to prefix the pull-request title with "[CI:DOCS]". That will prevent functional tests from running and save time and energy.

Finally, be sure to sign commits with your real name. Since by opening
a PR you already have commits, you can add signatures if needed with
something like `git commit -s --amend`.
-->

#### Does this PR introduce a user-facing change?

<!--
If no, just write `None` in the release-note block below. If yes, a release note
is required: Enter your extended release note in the block below. If the PR
requires additional action from users switching to the new release, include the
string "action required".

For more information on release notes, please follow the Kubernetes model:
https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
None
```

This PR clarifies things a bit. (It could probably be improved)

TomSweeneyRedHat commented on docs/source/markdown/podman-generate-systemd.1.md:
```diff
@@ -14,6 +14,8 @@ Generating unit files for a pod requires the pod to be created with an infra con
 
 _Note: If you use this command with the remote client, including Mac and Windows (excluding WSL2) machines, you would still have to place the generated units on the remote system.  Moreover, please make sure that the XDG_RUNTIME_DIR environment variable is set.  If unset, you may set it via `export XDG_RUNTIME_DIR=/run/user/$(id -u)`._
 
+_Note: The generated `podman run` command contains an `--sdnotify` option with the value taken from the container, unless the container value was set to __ignore__, then the value __conmon__ is used. If the container does not have any explicitly set value, the value __conmon__ is used._
```
```suggestion
_Note: The generated `podman run` command contains an `--sdnotify` option with the value taken from the container. If the container does not have any explicitly set value or the value is set to __ignore__, the value __conmon__ is used._
```

eriksjolund commented:
I made this a draft as there are more things to fix.


I noticed that
https://github.com/containers/podman/blob/main/docs/source/markdown/podman-generate-systemd.1.md#generate-systemd-unit-file-for-a-container-with---new-flag
has an old example where the generated output contains

```
Type=forking
```
Recent Podman versions generate 
```
Type=notify
```

eriksjolund commented:
I'll do a separate PR for fixing the very old examples. Let's keep this PR specific.

I incorporated some text from the [comment](https://github.com/containers/podman/issues/15029#issuecomment-1192244755) by @vrothberg.

Because of that, I added 
```
 Co-authored-by: Valentin Rothberg <vrothberg@redhat.com>
```
to the commit message. (I'm not sure what the correct procedure is)

vrothberg reviewed (commented):
/lgtm hold

@Luap99 mind having a final look?

vrothberg commented:
Thanks, @eriksjolund !

mheon commented:
/approve

Luap99 commented:
/lgtm
